### PR TITLE
Fix flytekit setup

### PR DESCRIPTION
**Pull request by cosmicBboy**
# TL;DR
- fixes some plugin unit tests
- updates linting make target to report all mypy errors
- adds plugin test suite to the github actions CI
- updates the contributing docs with more instructions on development environment setup

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: plugins/kftensorflow/flytekitplugins/kftensorflow/task.py [hunk 1]
```diff
@@ -53,9 +53,7 @@ def __init__(self, task_config: TfJob, task_function: Callable, **kwargs):
             task_type=self._TF_JOB_TASK_TYPE,
             task_config=task_config,
             task_function=task_function,
-            requests=task_config.per_replica_requests,
-            limits=task_config.per_replica_limits,
-            **kwargs
+            **{**kwargs, "requests": task_config.per_replica_requests, "limits": task_config.per_replica_limits}
```
Comment (wild-endeavor) [inline plugins/kftensorflow/flytekitplugins/kftensorflow/task.py:56]: not sure I understand this.  isn't the result the same?

**Review by wild-endeavor — APPROVED** [review]
approve but have a comment.

File: docs/source/contributing.rst [hunk 2]
```diff
@@ -18,9 +18,12 @@ We recommend using a virtual environment to develop flytekit. Inside the top lev
     source ~/.virtualenvs/flytekit/bin/activate
     make setup
     pip install -e .
+    pip install gsutil
```
Comment (wild-endeavor) [inline docs/source/contributing.rst:21]: probably also awscli right?

**Review by wild-endeavor — APPROVED** [review]

File: plugins/kftensorflow/flytekitplugins/kftensorflow/task.py (hunk 1 above)
Comment (cosmicBboy) [reply plugins/kftensorflow/flytekitplugins/kftensorflow/task.py:56]: this will override the `requests` and `limits` value in `kwargs` with whatever is in `task_config`. before the diff it would raise a `SyntaxError: keyword argument repeated` error because those two args are being passed explicitly and also unpacked via `kwargs`.

Comment (cosmicBboy) [reply plugins/kftensorflow/flytekitplugins/kftensorflow/task.py:56]: error might have cropped up because the `@task` function signature changed recently #401 and now both these args are being forwarded into the task class

**Review by wild-endeavor — APPROVED** [review]
